backend/internal/service/handler/media/get_media.go

```diff
@@ -9,3 +10,19 @@ func (h *Handler) GetMediaByName(c *fiber.Ctx) error {
 	medias, _ := h.mediaRepository.GetMediaByName(c.Context(), name)
 	return c.Status(fiber.StatusOK).JSON(medias)
 }
+
+func (h *Handler) GetMedia(c *fiber.Ctx) error {
+	// Check for the "type" query parameter
+	mediaType := c.Query("sort")
+	fmt.Println(mediaType)
+	if mediaType == "recent" {
+		// Call the method to get recent media if the "type" query parameter is "recent"
+		medias, err := h.mediaRepository.GetMediaByDate(c.Context())
+		if err != nil {
+			return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"error": "Failed to fetch recent media"})
+		}
+		return c.Status(fiber.StatusOK).JSON(medias)
+	}
+
+	return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid type query parameter"})
```

Comment:
I think we might want to handle this case differently (return _some_ media rather than throw), since query parameters should be treated as optional and not require an error if invalid or not provided. Right now though I'm not totally sure which media makes the most sense to return so this can be a future ticket!

Reply:
Yeah, the logic that I was thinking would be to return "recommended media" if that's something we were doing? If not, maybe media that was shared with the user could be returned by this endpoint? The error was a temporary fix of course.

Reply:
these are both great ideas i'm taking notes 📝